numpy/core/tests/test_api.py

```diff
@@ -289,6 +290,14 @@ class MyNDArray(np.ndarray):
     a = np.array(1000, dtype='i4')
     assert_raises(TypeError, a.astype, 'U1', casting='safe')
 
+@pytest.mark.parametrize("t",
+    np.sctypes['uint'] + np.sctypes['int'] + np.sctypes['float']
+)
+def test_array_astype_warning(t):
+    # test ComplexWarning when casting from complex to float or int
+    a = np.array(10, dtype=np.complex)
+    assert_raises(np.ComplexWarning, a.astype, t)
```

Comment:
Before we forget it. This should be `assert_warns` we only raise warnings during testing the development version, not in release mode (although I am not quite sure if pytest nowadays can be set up to handle that part). You may need a warnings context.

Reply:
Thanks for your reply. I have changed `assert_raises` to `assert_warns`.